mon-pix/app/services/locale.js

```diff
@@ -19,36 +23,79 @@ export default class LocaleService extends Service {
   @service dayjs;
   @service metrics;
 
+  get supportedLocales() {
+    return SUPPORTED_LOCALES;
+  }
+
+  get currentLocale() {
+    return this.intl.primaryLocale;
+  }
+
+  get acceptLanguageHeader() {
+    if (this.currentDomain.isFranceDomain) return FRENCH_FRANCE_LOCALE;
+    return this.currentLocale;
+  }
+
+  get availableLanguagesForSwitcher() {
+    const FRENCH_LANGUAGE = 'fr';
+    const options = Object.entries(VISIBLE_LANGUAGES)
+      .filter(([_, config]) => config.languageSwitcherDisplayed)
+      .map(([key, config]) => ({ label: config.value, value: key }));
+
+    return options.sort((option) => (option.value === FRENCH_LANGUAGE ? -1 : 1));
+  }
+
   isSupportedLocale(locale) {
     try {
       const localeCanonicalName = Intl.getCanonicalLocales(locale)?.[0];
-      return SUPPORTED_LOCALES.some((supportedLocale) => localeCanonicalName == supportedLocale);
+      return this.supportedLocales.some((supportedLocale) => localeCanonicalName == supportedLocale);
     } catch {
       return false;
     }
   }
 
-  handleUnsupportedLanguage(language) {
+  setLocale(locale) {
```

Comment:
Pour coller au vocabulaire métier (ubiquitous language) « locale courante » (`currentLocale`) et pour être cohérent avec le getter `currentLocale` le renommage suivant ne serait-il pas préférable ?

```suggestion
  setCurrentLocale(locale) {
```